### PR TITLE
Add functional tests /firefox/android/ page

### PR DESCRIPTION
**Pull request by alexgibson**

### CODE REVIEW

**Comment by alexgibson** [comment]
@davehunt r?

Note: I'm waiting on #3534 to merge before writing tests for the email form, but I'm not going to block on reviewing/merging these tests for the accordion.

**Comment by alexgibson** [comment]
@davehunt bump?

File: tests/functional/firefox/test_android.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    sections = page.customize_sections
+    for i in range(len(sections) - 1):
+        assert sections[i].is_displayed
+        page.show_next_customize_section()
+        assert not sections[i - 1].is_displayed
+    for i in range(len(sections) - 1, 0, -1):
+        assert sections[i].is_displayed
+        page.show_previous_customize_section()
+        assert not sections[i].is_displayed
+    assert sections[0].is_displayed
```
Comment (davehunt) [inline tests/functional/firefox/test_android.py]: I notice that the next/previous buttons work when you're on the first/last item to switch between them. Do we want to check this functionality too?

File: tests/functional/firefox/test_android.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+@pytest.mark.viewport('mobile')
+def test_mobile_accordion(base_url, selenium):
+    page = AndroidPage(base_url, selenium).open()
+    sections = page.customize_sections
+    assert not sections[0].is_displayed
+    for i in range(len(sections)):
+        sections[i].show_detail()
+        assert sections[i].is_displayed
+    for section in sections:
+        assert section.is_displayed
```
Comment (davehunt) [inline tests/functional/firefox/test_android.py]: You could replace this with:

```python
assert all(section.is_displayed for section in sections)
```

File: tests/functional/firefox/test_android.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    for i in range(len(sections) - 1, -1, -1):
+        sections[i].hide_detail()
+        assert not sections[i].is_displayed
+    for section in sections:
+        assert not section.is_displayed
```
Comment (davehunt) [inline tests/functional/firefox/test_android.py]: As above, but `assert not any(...)`

File: tests/pages/firefox/android.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                self.find_elements(self._customize_locator)]
+
+    def show_next_customize_section(self):
+        section = next(s for s in self.customize_sections if s.is_displayed)
+        self.scroll_element_into_view(self._next_button_locator).click()
+        self.wait.until(lambda s:
+            self.is_next_enabled and self.is_previous_enabled and not section.is_displayed)
+
+    def show_previous_customize_section(self):
+        section = next(s for s in self.customize_sections if s.is_displayed)
```
Comment (davehunt) [inline tests/pages/firefox/android.py]: You could abstract this into a `current_customize_section` property for less duplication.

**Comment by davehunt** [comment]
Comments are just recommendations, looks good! r+

**Comment by alexgibson** [comment]
Thanks for the suggestions @davehunt :+1:

**Pull request merged**